tests/tuner/test_lr_finder.py

```diff
@@ -295,3 +295,47 @@ def __init__(self, learning_rate=0.1, batch_size=2):
 
     assert lr != before_lr
     assert isinstance(bs, int)
+
+
+def test_lr_candidates_between_min_and_max(tmpdir):
+    """Test that learning rate candidates are between min_lr and max_lr."""
+    class TestModel(BoringModel):
+        def __init__(self, learning_rate=0.1):
+            super().__init__()
+            self.save_hyperparameters()
+
+    model = TestModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+    )
+
+    lr_min = 1e-8
+    lr_max = 1.0
+    lr_finder = trainer.tuner.lr_find(
+        model,
+        max_lr=lr_min,
+        min_lr=lr_max,
+    )
+    lr_candidates = lr_finder.results["lr"]
+    assert all([lr_min <= lr <= lr_max for lr in lr_candidates])
+
+
+def test_lr_finder_ends_before_num_training(tmpdir):
+    """Tests learning rate finder ends before `num_training` steps."""
+    class TestModel(BoringModel):
+        def __init__(self, learning_rate=0.1):
+            super().__init__()
+            self.save_hyperparameters()
+
+        def training_step_end(self, outputs):
+            assert self.global_step < num_training
```

Comment:
Unrelated to the changes in this PR, but a few questions about this

- Is the global_step increased during tuning?
- Is the optimizer state update during tuning?
- if yes, should it?

Reply:
> Is the global_step increased during tuning?

Yes, but it gets reset after tuning [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/013756404be0b665f393955f731665e2c09f8a10/pytorch_lightning/tuner/lr_finder.py#L191).

> Is the optimizer state update during tuning?

Yes, It seems so when fitting a model with various learning rates [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/479122d17a05e075c6cb02a5f4af03888c36abd9/pytorch_lightning/tuner/lr_finder.py#L179), but this state also gets reset after tuning.

> if yes, should it?

As far as I understand [the original paper](https://arxiv.org/pdf/1506.01186.pdf), I think it should.

cc: @SkafteNicki

Reply:
agree with @akihironitta observations